utils/common/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -133,6 +136,14 @@ fun File.unpack7Zip(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = {
 
             val target = targetDirectory.resolve(entry.name)
 
+            if (!target.canonicalFile.startsWith(targetDirectory)) {
```

Comment:
Would it make sense to compare this with `targetDirectory.canonicalFile` instead?
```suggestion
            if (!target.canonicalFile.startsWith(targetDirectory.canonicalFile)) {
```
(respectively evaluate `targetDirectory.canonicalFile` once outside the loop)

Otherwise I assume this check would fail if the `targetDirectory` value is not the canonical file path, even though the archive entry name is valid. Same applies to the other checks introduced by this PR as well.

(Sorry if this comment here is a bit intrusive)

Reply:
> Would it make sense to compare this with `targetDirectory.canonicalFile` instead?

Yes, you're right. Fixed.

> Sorry if this comment here is a bit intrusive

No worries, and thanks for the review!